pkg/server/server.go

```diff
@@ -686,10 +687,15 @@ func (s *ProxyServer) Connect(stream agent.AgentService_ConnectServer) error {
 				return
 			}
 
-			if s.warnOnChannelLimit && len(recvCh) >= xfrChannelSize {
+			select {
+			case recvCh <- in: // Send didn't block, carry on.
+			default: // Send blocked; record it and try again.
 				klog.V(2).InfoS("Receive channel on Connect is full", "agentID", agentID, "serverID", s.serverID)
```

Comment:
I think "on Connect" here is a bit obscure. Might be more clear if we referred to is as "forwarding traffic from the agent"?